File: backend/internal/service/user_service.go
package service

import (
	"api-aggregator/backend/internal/models"
	"api-aggregator/backend/internal/repository"
	"context"
	"errors"
	"fmt"
)

var (
	ErrUserNotFound = errors.New("user not found")
	ErrInvalidPage  = errors.New("invalid page parameters")
)

type UserService struct {
	userRepo *repository.UserRepository
}

func NewUserService(userRepo *repository.UserRepository) *UserService {
	return &UserService{
		userRepo: userRepo,
	}
}

// GetUsersRequest represents a request to get users list
type GetUsersRequest struct {
	Page     int `form:"page" binding:"omitempty,min=1"`
	PageSize int `form:"page_size" binding:"omitempty,min=1,max=100"`
}

// GetUsersResponse represents a paginated users response
type GetUsersResponse struct {
	Users    []*models.User `json:"users"`
	Total    int64          `json:"total"`
	Page     int            `json:"page"`
	PageSize int            `json:"page_size"`
}

// UpdateUserStatusRequest represents a request to update user status
type UpdateUserStatusRequest struct {
	Status string `json:"status" binding:"required,oneof=active inactive banned"`
}

// UpdateUserQuotaRequest represents a request to update user quota
type UpdateUserQuotaRequest struct {
	Quota int64 `json:"quota" binding:"required,min=0"`
}

// GetUsers returns a paginated list of users
func (s *UserService) GetUsers(ctx context.Context, req *GetUsersRequest) (*GetUsersResponse, error) {
	// Set default values
	if req.Page == 0 {
		req.Page = 1
	}
	if req.PageSize == 0 {
		req.PageSize = 10
	}

	// Validate parameters
	if req.Page < 1 || req.PageSize < 1 || req.PageSize > 100 {
		return nil, ErrInvalidPage
	}

	users, total, err := s.userRepo.FindAll(ctx, req.Page, req.PageSize)
	if err != nil {
		return nil, fmt.Errorf("failed to get users: %w", err)
	}

	return &GetUsersResponse{
		Users:    users,
		Total:    total,
		Page:     req.Page,
		PageSize: req.PageSize,
	}, nil
}

// GetUserByID returns a user by ID
func (s *UserService) GetUserByID(ctx context.Context, id uint) (*models.User, error) {
	user, err := s.userRepo.FindByID(ctx, id)
	if err != nil {
		return nil, fmt.Errorf("failed to get user: %w", err)
	}
	if user == nil {
		return nil, ErrUserNotFound
	}
	return user, nil
}

// UpdateUserStatus updates a user's status
func (s *UserService) UpdateUserStatus(ctx context.Context, id uint, status string) error {
	// Check if user exists
	user, err := s.userRepo.FindByID(ctx, id)
	if err != nil {
		return fmt.Errorf("failed to get user: %w", err)
	}
	if user == nil {
		return ErrUserNotFound
	}

	// Update status
	if err := s.userRepo.UpdateStatus(ctx, id, status); err != nil {
		return fmt.Errorf("failed to update user status: %w", err)
	}

	return nil
}

// UpdateUserQuota updates a user's quota
func (s *UserService) UpdateUserQuota(ctx context.Context, id uint, quota int64) error {
	// Check if user exists
	user, err := s.userRepo.FindByID(ctx, id)
	if err != nil {
		return fmt.Errorf("failed to get user: %w", err)
	}
	if user == nil {
		return ErrUserNotFound
	}

	// Update quota
	if err := s.userRepo.UpdateQuota(ctx, id, quota); err != nil {
		return fmt.Errorf("failed to update user quota: %w", err)
	}

	return nil
}

File: admin/src/services/userService.ts
import { apiClient } from '../lib/api';
import { User } from '../types';

export interface UsersListResponse {
  users: User[];
  total: number;
  page: number;
  page_size: number;
}

export interface UpdateUserStatusRequest {
  status: string;
}

export interface UpdateUserQuotaRequest {
  quota: number;
}

export const userService = {
  // 获取用户列表
  getUsers: async (params?: {
    page?: number;
    page_size?: number;
    search?: string;
    status?: string;
  }): Promise<UsersListResponse> => {
    const response = await apiClient.get<UsersListResponse>('/admin/users', {
      params,
    });
    return response.data;
  },

  // 获取用户详情
  getUserById: async (id: number): Promise<User> => {
    const response = await apiClient.get<User>(`/admin/users/${id}`);
    return response.data;
  },

  // 更新用户状态
  updateUserStatus: async (
    id: number,
    data: UpdateUserStatusRequest
  ): Promise<User> => {
    const response = await apiClient.put<User>(
      `/admin/users/${id}/status`,
      data
    );
    return response.data;
  },

  // 更新用户额度
  updateUserQuota: async (
    id: number,
    data: UpdateUserQuotaRequest
  ): Promise<User> => {
    const response = await apiClient.put<User>(
      `/admin/users/${id}/quota`,
      data
    );
    return response.data;
  },

  // 禁用用户
  disableUser: async (id: number): Promise<User> => {
    return userService.updateUserStatus(id, { status: 'disabled' });
  },

  // 启用用户
  enableUser: async (id: number): Promise<User> => {
    return userService.updateUserStatus(id, { status: 'active' });
  },
};
